libraries/browser-tracker-core/test/browser_props.test.ts

```diff
@@ -10,4 +10,18 @@ describe('Browser props', () => {
     const testFractionalDimensions = '100.2x100.1';
     expect(floorDimensionFields(testFractionalDimensions)).toEqual('100x100');
   });
+
+  describe('#getBrowserProperties', () => {
+    describe('with undefined document', () => {
+      beforeAll(() => {
+        // @ts-expect-error
+        document = undefined;
+      });
+  
+      it('does not invoke the resize observer if the document is null', () => {
+        const browserProperties = getBrowserProperties();
+        expect(browserProperties).not.toEqual(null);
+      });
+    });
+  });
```

Comment:
**Note:** Normally, I would have included tests where `document.body` and `document.documentElement` were undefined as well - these caused compilation errors, so I have removed them.